### ISSUE TITLE
plot broken with new ggplot2

### ISSUE THREAD

**Issue by thibautjombart** [reconhub/incidence#34]
Dear Thibaut Jombart,

This is an automated email to let you know that:

* A new version of ggplot2 is ready to go to CRAN. ggplot2 is
  currently at version 2.2.1.9000 and will become 2.3.0 upon release.

* incidence uses ggplot2 and has problems with the new version.

* We plan to submit ggplot2 to CRAN on June 25.

I need your help to keep incidence and ggplot2 working together
smoothly. In the next one month, can you please:

1. Read about the changes to ggplot2 at
   https://github.com/tidyverse/ggplot2/blob/master/NEWS.md#ggplot2-2219000.
   This page includes a list of breaking changes, the reasoning behind
   them, and to how to update your code.

2. Carefully inspect the failing checks listed at the bottom of this email.

3. For each failing check, either update your package, or tell me
   that I have a bug. If you have made changes to your package, please
   submit an update to CRAN before June 25.

If you have discovered a bug in ggplot2, please file an issue (ideally
with a small reprex that illustrates the problem) at
https://github.com/tidyverse/ggplot2/issues. If you're not sure
whether or not you've found a bug, please an issue and we'll help you
figure it out.

Thanks,

Hadley Wickham
```
== CHECK RESULTS ========================================

*   checking examples ... ERROR
    ```
    ...
    +  ## plot data and model (recommended)
    +  plot(i.7, fit=f)
    +  plot(i.7[1:25], fit=f)
    +
    +
    + ## EXAMPLE WITH 2 PHASES
    +  ## specifying the peak manually
    +  f2 <- fit(i.7, split=as.Date("2014-10-15"))
    +  f2
    +  plot(i.7, fit=f2)
    +
    + ## finding the best 'peak' date
    + f3 <- fit_optim_split(i.7)
    + f3
    + plot(i.7, fit=f3$fit)
    + }
    Loading required package: outbreaks
    Error in as.Date.default(x.major_source, origin = "1970-01-01") :
      do not know how to convert 'x.major_source' to class “Date”
    Calls: plot -> plot.incidence -> as.Date -> as.Date.default
    Execution halted
    ```

*   checking re-building of vignette outputs ... WARNING
    ```
    Error in re-building vignettes:
      ...
    Quitting from lines 80-86 (conversions.Rmd)
    Error: processing vignette 'conversions.Rmd' failed with diagnostics:
    do not know how to convert 'x.major_source' to class "Date"
    Execution halted
    ```

**Comment by thibautjombart** [comment]
Fixed via 031afccb85b42895d29cf11faa97e82ff5469680

Will need to time release with ggplot2

**Comment by thibautjombart** [comment]
Fixed using a switch at 5aeb036af66665037d966c57826d3c27b70d404f
